src/LogicTasks/Semantics/Resolve.hs

```diff
@@ -160,64 +187,32 @@ partialGrade ResolutionInst{..} sol = do
       pure ()
     )
 
-  preventWithHint (not $ null noResolveSteps)
-    (translate $ do
-      german "Alle Schritte sind gültig?"
-      english "All steps are valid?"
-    )
-    (paragraph $ do
-      translate $ do
-        german "Mindestens ein Schritt ist kein gültiger Resolutionsschritt. "
-        english "At least one step is not a valid resolution step. "
-      itemizeM $ map (text . show) noResolveSteps
-      pure ()
-    )
+  when showFeedbackOnPartialGrade $ do
+    gradeSteps sol clauses
 
-  prevent checkEmptyClause $
-    translate $ do
-      german "Letzter Schritt leitet die leere Klausel ab?"
-      english "The last step derives the empty clause?"
   pure ()
   where
     checkMapping = correctMapping sol $ baseMapping clauses
     steps =  replaceAll sol $ baseMapping clauses
-    checkEmptyClause = null steps || not (isEmptyClause $ third3 $ last steps)
     availLits = unions (map (fromList . literals) clauses)
     stepLits (c1,c2,r) = toList $ unions $ map (fromList . literals) [c1,c2,r]
     wrongLitsSteps = filter (not . all (`member` availLits) . stepLits) steps
-    noResolveSteps = filter (\(c1,c2,r) -> maybe True (\x ->
-      fromJust (resolve c1 c2 x) /= r) (resolvableWith c1 c2)) steps
-
-
 
 completeGrade :: OutputMonad m => ResolutionInst -> [ResStep] -> LangM m
-completeGrade ResolutionInst{..} sol =
-    case applySteps clauses steps of
-        Nothing -> refuse $ indent $ do
-          translate $ do
-            german "In mindestens einem Schritt werden Klauseln resolviert, die nicht in der Formel sind oder noch nicht abgeleitet wurden."
-            english "In at least one step clauses are used, that are not part of the original formula and are not derived from previous steps."
+completeGrade ResolutionInst{..} sol = do
+    unless showFeedbackOnPartialGrade $ do
+      gradeSteps sol clauses
```

Comment:
Hier würde wieder ein "zu früher Abbruch" erfolgen, richtig?

Eventuell lässt sich das mit `recoverFrom` oder `recoverWith` aus `output-monad` geeignet vermeiden.

Oder das `gradeSteps` wird in `completeGrade` einfach nicht verwendet? (Also es passiert dann entweder in `partialGrade` oder gar nicht.)

Reply:
Allerdings müsste dann (vielleicht auch sowieso) unten noch ein explizites `reject` in denjenigen Fällen erfolgen, wo `isCorrect == False`. Das vormalige `refuse` in den Fällen `applied == Nothing` und `not (any isEmptyClause (fromJust applied))` ist aktuell ja nicht mehr vorhanden.

Reply:
Mit den neusten Änderungen sollte es nun richtig abbrechen. Mit `recoverFrom` und `recoverWith` konnte ich rein anhand der Definition nichts anfangen.

Reply:
(The behavior of `recoverFrom` should be that it kind of "catches" a `refuse/reject`.  That is, `recoverFrom (do ... reject)` would execute the do-block, print all messages the `...` part prints, but then ignore that there is a `reject` involved. Thus being able to do additional printing and checking. But the way you changed the code in `Resolve` now should also have the desired effect.)